### ISSUE TITLE
[BUG] Disconnecting while in a dungeon removes from group, players remain in dungeon map

### ISSUE THREAD

**Issue by MarcatoSound** [DRE2N/DungeonsXL#329]
Hey there,

Sorry to report a vague and uninformative bug, but on version 0.16 there's been an unidentified bug where players are literally incapable of exiting a dungeon, not by kicking them from it, by teleporting them, or by triggering an 'end' sign. The plugin also failed to reload while the problem persisted. I've been trying to replicate it to no avail thus far, however did discover that I was able to rescue a player through the following steps:

1. Having the player disconnect from the server
2. Removing their player file in the DXL plugins folder (saving a back-up offline)
3. Reloading the plugin
4. Re-uploading their player file (to hopefully return their inventory to them, which it did.)
5. Reloading the plugin again
6. Having the player connect to the server again.

This may just be a case of a hard-reset being performed on the player, but nevertheless since the cause has yet to be discovered, I thought I'd share what little I've found.

Bonus info: Even after losing all their lives, the player continued to re-spawn at the last checkpoint in the dungeon.

Myself and my team are going to have a mass-testing session tomorrow in an attempt to reproduce the error. I will update this thread as soon as more info becomes available.

**Comment by MarcatoSound** [comment]
UPDATE: One of my team members discovered how to replicate the bug and its details.

Upon disconnecting, and reconnecting, players are removed from their 'group' but remain in the dungeon when they reconnect. After that no commands to exit work any longer, including completing the dungeon via 'end' sign.

**Comment by MarcatoSound** [comment]
UPDATE: We've discovered a fix to the bug, however it appears that the original bug is caused by the recent changes to the 'timeUntilKickOfflinePlayer' option.


Prior to implementing the development build, it was:
`timeUntilKickOfflinePlayer: 120`
Upon updating to 0.16, after fighting with it for a while, the dungeons nearly became unplayable due to this bug, but we attempted changing it to "0" to match the default value implemented in 0.16, and discovered this completely eliminated the problem.

Perhaps there's a bug in how DXL handles the offline player kicking?
